pkg/operator/csi/csidefaultstorageclasscontroller/csi_default_storageclass_controller_test.go

```diff
@@ -0,0 +1,322 @@
+					makeFakeScInstance("test-sc-2", false),
+				},
+			},
+			applyDefaultSc: true,
+			expectedObjects: testObjects{
+				storageClasses: []*storagev1.StorageClass{
+					makeFakeScInstance("test-sc", false),
+					makeFakeScInstance("test-sc-2", false),
+					fakeAssetFuncToScObject(fakeAssetFuncFactory(true)),
+				},
+			},
+		},
+		{
+			name: "test sync default sc - prior default is set - applied is not default",
+			initialObjects: testObjects{
+				storageClasses: []*storagev1.StorageClass{
+					makeFakeScInstance("test-sc", true),
+					makeFakeScInstance("test-sc-2", false),
+				},
+			},
+			applyDefaultSc: true,
+			expectedObjects: testObjects{
+				storageClasses: []*storagev1.StorageClass{
+					makeFakeScInstance("test-sc", true),
+					makeFakeScInstance("test-sc-2", false),
+					fakeAssetFuncToScObject(fakeAssetFuncFactory(false)),
+				},
+			},
+		},
+		{
+			name: "test sync default sc - no prior sc - applied is default",
+			initialObjects: testObjects{
+				storageClasses: []*storagev1.StorageClass{},
+			},
+			applyDefaultSc: true,
+			expectedObjects: testObjects{
+				storageClasses: []*storagev1.StorageClass{
+					fakeAssetFuncToScObject(fakeAssetFuncFactory(true)),
+				},
+			},
+		},
+		{
+			name: "test sync default sc - no prior default set - sync must not rewrite to default",
+			initialObjects: testObjects{
+				storageClasses: []*storagev1.StorageClass{
+					makeFakeScInstance("test-sc", false),
+					makeFakeScInstance("test-sc-2", false),
+					fakeAssetFuncToScObject(fakeAssetFuncFactory(false)),
+				},
+			},
+			applyDefaultSc: true,
+			expectedObjects: testObjects{
+				storageClasses: []*storagev1.StorageClass{
+					makeFakeScInstance("test-sc", false),
+					makeFakeScInstance("test-sc-2", false),
+					fakeAssetFuncToScObject(fakeAssetFuncFactory(false)),
+				},
+			},
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			// Initialize
+			ctx := newTestContext(test, t)
+
+			// Act
+			_ = ctx.controller.Sync(context.TODO(), factory.NewSyncContext(controllerName, events.NewInMemoryRecorder(operandName)))
+
+			// Assert
+			actualSCList, _ := ctx.kubeClient.StorageV1().StorageClasses().List(context.TODO(), metav1.ListOptions{})
+			actualSCs := map[string]*storagev1.StorageClass{}
+			for i := range actualSCList.Items {
+				sc := &actualSCList.Items[i]
+				actualSCs[sc.Name] = sc
+			}
+			expectedSCs := map[string]*storagev1.StorageClass{}
+			for _, sc := range test.expectedObjects.storageClasses {
+				expectedSCs[sc.Name] = sc
+			}
+
+			for name, expectedSC := range expectedSCs {
+				actualSC, found := actualSCs[name]
+				if !found {
+					t.Errorf("Expected StorageClass not found: %s", name)
+					continue
+				}
+				if !equality.Semantic.DeepEqual(expectedSC, actualSC) {
+					t.Errorf("Unexpected StorageClass %+v content:\n%s", name, cmp.Diff(expectedSC, actualSC))
+				}
+				delete(expectedSCs, name)
```

Comment:
Why do you delete the SC? It almost seems that you wanted to check if the list is empty after the loop.

Reply:
not needed - removing